tests/Avalonia.Controls.UnitTests/Templates/TemplateExtensionsTests.cs

```diff
@@ -44,5 +44,28 @@ public void GetTemplateChildren_Should_Not_Return_Nested_Template_Controls()
 
             Assert.Equal(new[] { "border1", "inner", "border4" }, result);
         }
+
+        [Fact]
+        public void ControlTemplate_FindName_Finds_Named_Control()
+        {
+            var target = new TestTemplatedControl
+            {
+                Template = new FuncControlTemplate<TestTemplatedControl>(parent =>
+                    new Border
+                    {
+                        Name = "PART_Border",
+                        Child = new Canvas
+                        {
+                            Name = "PART_Canvas",
+                        }
+                    })
+            };
+
+            target.ApplyTemplate();
+
+            var result = target.Template.FindName("PART_Canvas", target);
```

Comment:
As the comment above, I do not know whether the "PART_Canvas" name not found or the "PART_Canvas" is not a canvas when `result` is null.